File: scripts/analysis/send_custom_message.py
#!/usr/bin/env python3
"""
Send a custom message to a specific participant.

Python port of scripts/send-custom-message.ts.

Environment variables:
  PROLIFIC_API_TOKEN  – Prolific API token (required)
  STUDY_ID            – Prolific study ID (required)
  PID                 – Participant ID to message (required)
  MESSAGE             – Message body to send (required)
  DRY_RUN             – When "false", send live (default: true)
"""

import os
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).parent))

from tabs_api import prolific_send_message


def _require_env(name: str) -> str:
    value = os.environ.get(name)
    if not value:
        print(f"Error: {name} is required", file=sys.stderr)
        sys.exit(1)
    return value


def main():
    api_token = _require_env("PROLIFIC_API_TOKEN")
    study_id = _require_env("STUDY_ID")
    pid = _require_env("PID")
    message = _require_env("MESSAGE")
    dry_run = os.environ.get("DRY_RUN", "true").strip().lower() != "false"

    print(f"PID: {pid}")
    print(f"Message length: {len(message)} chars")
    print(f"Mode: {'DRY RUN' if dry_run else 'LIVE'}")
    print()
    print(f"Message: {message}")
    print()

    if dry_run:
        print("DRY RUN - message not sent")
    else:
        prolific_send_message(study_id, pid, message, api_token)
        print("Message sent successfully")


if __name__ == "__main__":
    main()

File: scripts/archive/send-custom-message.ts
/**
 * Send a custom message to a specific participant.
 *
 * Environment:
 *   PROLIFIC_API_TOKEN  – Prolific API token (required)
 *   STUDY_ID            – Prolific study ID (required)
 *   PID                 – Participant ID to message (required)
 *   MESSAGE             – Message body to send (required)
 *   DRY_RUN             – When "false", send live (default: true)
 */

import { sendMessage } from '../src/lib/prolific-api'

async function main() {
  const token = process.env.PROLIFIC_API_TOKEN
  if (!token) {
    console.error('PROLIFIC_API_TOKEN required')
    process.exit(1)
  }
  const studyId = process.env.STUDY_ID
  if (!studyId) {
    console.error('STUDY_ID required')
    process.exit(1)
  }
  const pid = process.env.PID
  if (!pid) {
    console.error('PID required')
    process.exit(1)
  }
  const message = process.env.MESSAGE
  if (!message) {
    console.error('MESSAGE required')
    process.exit(1)
  }
  const dryRun = (process.env.DRY_RUN ?? 'true').toLowerCase() !== 'false'

  console.log(`PID: ${pid}`)
  console.log(`Message length: ${message.length} chars`)
  console.log(`Mode: ${dryRun ? 'DRY RUN' : 'LIVE'}`)
  console.log('')
  console.log(`Message: ${message}`)
  console.log('')

  if (dryRun) {
    console.log('DRY RUN - message not sent')
  } else {
    await sendMessage(studyId, pid, message, token)
    console.log('Message sent successfully')
  }
}

main().catch((e) => {
  console.error(e)
  process.exit(1)
})
